torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1275,9 +1275,10 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad):
            sample_inputs_func=sample_inputs_linalg_norm,
            aten_name='linalg_norm',
            skips=(
-               # TODO: remove this once `pow` is implemented for float16
-               #       and bfloat16 on CPU. Issue:
-               #       https://github.com/pytorch/pytorch/issues/50789
+               # TODO: remove this SkipInfo once these tests can pass.
+               # Although pow() has now been implemented for float16 & bfloat16,
+               # these tests are still failing, so skipping them for now.
+               # https://github.com/pytorch/pytorch/issues/48978#issuecomment-766698749
                SkipInfo('TestCommon', 'test_variant_consistency_jit',
```

Comment:
These tests were supposed to be unskipped after `pow()` would have been supported for `bfloat16` & `float16`, but both are failing, so I'm still skipping them for now.

Reply:
Installing magma libraries to check why CPU & CUDA have different behavior. The CUDA tests require PyTorch to be built with magma.

Reply:
Correct me if I'm wrong, but it doesn't look like this PR adds forward mode tests for float16 and bfloat16. Could you add tests for that? I'm wondering if eager could be slightly incorrect, which could possibly be causing the JIT error (although I'm not familiar with what `test_variant_consistency_jit_*` actually does).

Reply:
Hello @kurtamohler, thanks a lot for taking a look! Could you please let me know what tests you're referring to? Thanks again!

Reply:
It looks like there are a few different forward mode eager tests for `pow` in the `test` directory. You might have to look around to see if there are more (could use `$ git grep -w pow test` to help find them), but here are a couple that I found:

- `test_pow` in `test/test_binary_ufuncs.py`: https://github.com/pytorch/pytorch/blob/master/test/test_binary_ufuncs.py#L472
- Entries in `tensor_op_tests` in `test/test_torch.py` (note that the `get_all_fp_types()` call does _not_ include half dtypes): https://github.com/pytorch/pytorch/blob/master/test/test_torch.py#L6554

Reply:
Thanks again for your help!

Reply:
No problem, glad to help!

Reply:
@kurtamohler: Since `pow()` was already supported for CUDA for `float16` & `bfloat16`, please advise if I should only find those tests that are currently being skipped for CPU due to the hitherto lack of `pow()` support on CPU for these types. Apart from the tests you mentioned, I couldn't find any other tests that directly test `pow()`. 

As for the tests you pointed out (please only read the text in bold-face if you'd like to skim over, as this comment is quite verbose):

1.  `test_pow` in `test/test_binary_ufuncs.py`: 

This test is being run for the dtypes returned from `torch.testing.get_all_math_dtypes()`. For CUDA, it returns `float16` as well. For CPU, it returns neither `float16` nor `bfloat16` :
https://github.com/pytorch/pytorch/blob/febc76a5c69302c7d4b481632bcbd25e5d9b76ca/torch/testing/__init__.py#L339


**I can enable the test for CPU (`float16`, `bfloat16`) & CUDA(`bfloat16`).** All of them passed locally.

2. Entries in `tensor_op_tests` in `test/test_torch.py`

`torch.testing.get_all_fp_dtypes()`  actually returns `[torch.float32, torch.float64, torch.float16, torch.bfloat16]`.

These tests compare outputs of another device (CUDA for GPU) with a CPU-based implementation. When it comes to half types, they use `float32` instead of `float16` & `bfloat16`. To ascertain correctness, they compare for equality `float32` tensors with `float16` or `bfloat16` tensors that were computed on another device (GPU, for example), while allowing for some acceptable error-bounds. But if the other device is CPU itself, then very few ops are being tested. In fact, `pow()` wasn't even being tested for `float32` & `float64` on CPU, because testing the CPU implementation for `float32` & `float64` against itself wouldn't have made much sense!

**I can enable `pow()` related tests for `float16` & `bfloat16` for CPU.** However, that'd mean `float32` tensors would be compared for equality with `float16` & `bfloat16` tensors (with bounded error). That's how the CUDA implementation of `pow()` for `float16` & `bfloat16` is also being tested, so it should be fine. All such tests are passing for `pow()` locally. 



The other `pow()` tests I found don't need to be modified, as they're not testing for every floating point type:
1. `test_pow()` in `test/test_tensorexpr.py` uses `torch.pow()` but only for `float`.
2. `test_exp_pow()` in `test/test_tensorexpr.py` also uses `torch.pow()` but for `double`.

**Thanks again for your help!**

Reply:
Thanks for investigating!

> Apart from the tests you mentioned, I couldn't find any other tests that directly test pow().

If there are no other tests, that's probably fine. The existing ones seem to be robust enough, if I'm not mistaken.

> I can enable the test for CPU (float16, bfloat16) & CUDA(bfloat16). All of them passed locally.

Great!

> torch.testing.get_all_fp_dtypes() actually returns [torch.float32, torch.float64, torch.float16, torch.bfloat16].

Ah ok, I was wrong about that, sorry.

> I can enable pow() related tests for float16 & bfloat16 for CPU

I think that's probably a good idea

Reply:
Thanks again, @kurtamohler!

BTW, in the `Files Changed` tab on GitHub, I'm being shown diff between my last 2 commits. I find it strange that GitHub has no option for displaying the diff between the target-branch (`pytorch:master`) for the files changed in the PR. Currently, a reviewer (you, for example) would also see additional changes made in the main repo that I merged with a merge-tool (meld) locally, so reviewing might be inconvenient.

Please let me know if you'd recommend reverting my changes, rebasing & then adding those changes back in, so that you may only see the diff corresponding to the changes that I made.

Thanks!